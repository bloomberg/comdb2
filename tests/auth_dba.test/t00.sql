select 'disabled' as authentication_status;

select * from comdb2_users order by username;
select * from comdb2_tunables where name like '%dba%' order by name;

create table t1(i int)$$
insert into t1 values(1);
select * from t1;

-- test 'lock_dba_user'
grant op to 'dba';
revoke op to 'dba';
put password off for 'dba';
grant op to 'dba';
revoke op to 'dba';
revoke read on 't1' to 'dba';

put password 'root' for 'root'
grant op to 'root';
select * from comdb2_users;

-- login using 'root' user and enable authentication
set user 'root'
set password 'root'
select 'root' as current_user;
put authentication on

select 'enabled' as authentication_status;

select * from comdb2_users order by username;
select * from t1;
-- another op user must not still not be able to change 'dba' user permissions
grant op to 'dba';
revoke op to 'dba';
put password off for 'dba';
grant op to 'dba';
revoke op to 'dba';
revoke read on 't1' to 'dba';

-- login using 'dba' user
set user 'dba'
set password ''
select 'dba' as current_user;
put authentication off
put authentication on
put password off for 'root'

select * from comdb2_users order by username;

-- cleanup
put authentication off
drop table t1;
