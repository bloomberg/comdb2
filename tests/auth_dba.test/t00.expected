(authentication_status='disabled')
(username='dba', isOP='Y')
(name='create_dba_user', description='Automatically create 'dba' user if it does not exist already (Default: on)', type='BOOLEAN', value='ON', read_only='Y')
(name='lock_dba_user', description='When enabled, 'dba' user cannot be removed and its access permissions cannot be modified. (Default: off)', type='BOOLEAN', value='ON', read_only='Y')
(rows inserted=1)
(i=1)
[grant op to 'dba'] failed with rc 203 dba user is locked!
[revoke op to 'dba'] failed with rc 203 dba user is locked!
[put password off for 'dba'] failed with rc 203 dba user is locked!
[grant op to 'dba'] failed with rc 203 dba user is locked!
[revoke op to 'dba'] failed with rc 203 dba user is locked!
[revoke read on 't1' to 'dba'] failed with rc 203 dba user is locked!
(username='dba', isOP='Y')
(username='root', isOP='Y')
(current_user='root')
(authentication_status='enabled')
(username='dba', isOP='Y')
(username='root', isOP='Y')
(i=1)
[grant op to 'dba'] failed with rc 203 dba user is locked!
[revoke op to 'dba'] failed with rc 203 dba user is locked!
[put password off for 'dba'] failed with rc 203 dba user is locked!
[grant op to 'dba'] failed with rc 203 dba user is locked!
[revoke op to 'dba'] failed with rc 203 dba user is locked!
[revoke read on 't1' to 'dba'] failed with rc 203 dba user is locked!
(current_user='dba')
(username='dba', isOP='Y')
